test/Main.purs

```diff
@@ -799,18 +816,23 @@ main = do
   tokenParserCommaSepTest
   tokenParserCommaSep1Test
 
+  log "\nTESTS Haskell Style\n"
   haskellStyleTest
+  log "\nTESTS Java Style\n"
   javaStyleTest
 
-  case runParser "aa" p of
-    Right _ -> assert' "error: ParseError expected!" false
-    Left (ParseError message _) -> do
-      let messageExpected = "context1 context2 Expected \"b\""
-      assert' ("expected message: " <> messageExpected <> ", message: " <> message) (message == messageExpected)
-      logShow messageExpected
-  where
-  prependContext m' (ParseError m pos) = ParseError (m' <> m) pos
-  p = region (prependContext "context1 ") $ do
-    _ <- string "a"
-    region (prependContext "context2 ") $ do
-      string "b"
+  log "\nTESTS region\n"
+  ( let
+      prependContext m' (ParseError m pos) = ParseError (m' <> m) pos
+      p = region (prependContext "context1 ") $ do
+        _ <- string "a"
+        region (prependContext "context2 ") $ do
+          string "b"
+    in
+      case runParser "aa" p of
+        Right _ -> assert' "error: ParseError expected!" false
+        Left (ParseError message _) -> do
+          let messageExpected = "context1 context2 Expected \"b\""
+          assert' ("expected message: " <> messageExpected <> ", message: " <> message) (message == messageExpected)
+          logShow messageExpected
+  )
```

Comment:
The `p` was shadowing something else, so now those bindings are locally scoped.

Reply:
Sorry, I'm saying that, because you are in a `do` block, ending with a `let/in` like this is exactly the same as just using a `let` in the do block, with the last statement being the `runParser`. The parens don't do anything semantically. If inserting the parens, vs using a do/let results in different warnings, it's a bug in the compiler.

Reply:
Unless more tests are added after this block. Anyway I deleted the parens.